File: helpers/Dropdown.ts
import { Locator, Page } from '@playwright/test';

export class Dropdown {
  private container: Locator;

  /** Returns a single element */
  get value(): Locator {
    return this.container.locator('.react-select__value-container');
  }

  /** May return multiple elements */
  get values(): Locator {
    return this.container.locator('.react-select__single-value')
      .or(this.container.locator('.react-select__multi-value'));
  }

  constructor(page: Page, key: string) {
    const input = page.locator('.react-select__input-container').getByLabel(key);
    this.container = page.locator('.react-select__control').filter({has: input});
  }

  async selectOption(label: string) {
    await this.toggle();

    await this.container
      .locator('..') // xpath parent
      .locator('.react-select__option')
      .getByText(label, {exact: true})
      .click();
  }

  private async toggle() {
    await this.container.locator('.react-select__indicators').click();
  }
}

File: Itenium.Confac.Playwright/Helpers/Dropdown.cs
using Microsoft.Playwright;

namespace Itenium.Confac.Playwright.Helpers;

public class Dropdown
{
  private readonly ILocator _container;

  /// <summary>Returns a single element</summary>
  public ILocator Value => _container.Locator(".react-select__value-container");

  /// <summary>May return multiple elements</summary>
  public ILocator Values => _container.Locator(".react-select__single-value").Or(_container.Locator(".react-select__multi-value"));

  public Dropdown(IPage page, string key)
  {
    ILocator input = page.Locator(".react-select__input-container").GetByLabel(key);
    _container = page.Locator(".react-select__control").Filter(new LocatorFilterOptions() {Has = input});
  }

  public async Task SelectOption(string label)
  {
    await Toggle();

    await _container
      .Locator("..")
      .Locator(".react-select__option")
      .GetByText(label, new() { Exact = true })
      .ClickAsync();
  }

  private async Task Toggle()
  {
    await _container.Locator(".react-select__indicators").ClickAsync();
  }
}
